catalyst_voices/apps/voices/lib/common/formatters/date_formatter.dart

```diff
@@ -1,9 +1,25 @@
 import 'package:catalyst_voices_localization/catalyst_voices_localization.dart';
 import 'package:catalyst_voices_shared/catalyst_voices_shared.dart';
+import 'package:catalyst_voices_view_models/catalyst_voices_view_models.dart';
 import 'package:intl/intl.dart';
 
 /// A [DateTime] formatter.
 abstract class DateFormatter {
+  static String formatCampaignDetailDate(
+    VoicesLocalizations l10n,
+    DateTime date,
+    CampaignStage stage,
```

Comment:
i dont think `CampaignStage stage` logic should be here. Just simple formatting. l10n seems to be explicit to use case and not formatting itself.